### PR TITLE
block_explosives remove remnants when detached.

### LINKED ISSUE #75793: Explosive block does not consistently go away, allowing for infinite stacking of it on any turf. (Massive exploit/bug)
## Reproduction:

https://github.com/tgstation/tgstation/assets/62126254/a707dbd0-6fd9-4cdb-8798-75660bc90b0e

https://github.com/tgstation/tgstation/assets/62126254/c185bd45-a5fe-49a1-b2ff-7a3a5b319252

### PR DESCRIPTION
**Pull request by Pickle-Coding**
Removes the explosive resistance remnants on the turf(s) if blocks_explosives gets detached.
## About The Pull Request
Closes #75793

Also this is an untested webedit.
## Why It's Good For The Game
hhhhhhhhhhhhhhhhh
## Changelog
:cl:
fix: Fix movable explosive blockers leaving remnants of explosive protection on turfs after they get destroyed.
/:cl:

### CODE REVIEW

File: code/datums/elements/blocks_explosives.dm [hunk 1]
```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
+		var/atom/movable/target = target
+		if(length(target.locs) > 1)
+			for(var/atom/location as anything in target.locs)
+				unblock_loc(location, target.explosion_block)
+		else if(target.loc)
+			unblock_loc(location, target.explosion_block)
```
Comment (Pickle-Coding) [inline code/datums/elements/blocks_explosives.dm:29]:
```suggestion
			unblock_loc(target.loc, target.explosion_block)
```

File: code/datums/elements/blocks_explosives.dm [hunk 2]
```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
+		var/atom/movable/target = target
```
Comment (Pickle-Coding) [inline code/datums/elements/blocks_explosives.dm:24]:
```suggestion
		var/atom/movable/target = source
```

Comment (Pickle-Coding) [reply code/datums/elements/blocks_explosives.dm:24]: Silly me.

File: code/datums/elements/blocks_explosives.dm [hunk 3]
```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
+		var/atom/movable/target = source
+		if(length(target.locs) > 1)
+			for(var/atom/location as anything in target.locs)
+				unblock_loc(location, target.explosion_block)
+		else if(target.loc)
+			unblock_loc(target.loc, target.explosion_block)
```
Comment (LemonInTheDark) [inline code/datums/elements/blocks_explosives.dm:29]: won't work, detach is called when the element is explicitly removed.
this is still good to have, but to make this work with the issues you're looking at you'll need `ELEMENT_DETACH_ON_HOST_DESTROY` as an element flag

File: code/datums/elements/blocks_explosives.dm [hunk 4]
```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
```
Comment (LemonInTheDark) [inline code/datums/elements/blocks_explosives.dm:23]: this is unneeded, source cannot be anything but atom/movable here, so you can just drop this typecheck and the cast you do on the next line

Comment (Pickle-Coding) [reply code/datums/elements/blocks_explosives.dm:24]: Can I do the cast on the proc parameter without causing problems? I think the casting would be useful for code editors.

Comment (LemonInTheDark) [reply code/datums/elements/blocks_explosives.dm:24]: yep, that's fine byond won't care and only atoms should get there regardless

**Pull request merged**